### PR TITLE
[PHPUnit] Try catch to expectException

### LINKED ISSUE #591: [PHPUnit] Try/catch to setExpectedException()
```php
try {
	// ...
} catch (Throwable $exception) {
    $this->assertInstanceOf(RuntimeException::class, $e);
    $this->assertContains('There was an error executing the following script', $e->getMessage());
}
```

↓

```php
$this->expectException(RuntimeException::class);
$this->expectExceptionMessage('There was an error executing the following script');

// ...
```

### PR DESCRIPTION
**Pull request by TomasVotruba**
Closes #591

### CODE REVIEW

File: packages/PHPUnit/tests/Rector/TryCatchToExpectExceptionRector/TryCatchToExpectExceptionRectorTest.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+     * @dataProvider provideWrongToFixedFiles()
+     */
+    public function test(string $wrong, string $fixed): void
+    {
+        $this->doTestFileMatchesExpectedContent($wrong, $fixed);
+    }
+
+    public function provideWrongToFixedFiles(): Iterator
+    {
+        yield [__DIR__ . '/Wrong/wrong.php.inc', __DIR__ . '/Correct/correct.php.inc'];
```
Comment (TomasVotruba) [inline packages/PHPUnit/tests/Rector/TryCatchToExpectExceptionRector/TryCatchToExpectExceptionRectorTest.php]: Improve test cases with more try method calls and false positvites

**Comment by carusogabriel** [comment]
I’ll take a look this afternoon.

File: packages/PHPUnit/src/Rector/TryCatchToExpectExceptionRector.php [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    private $methodCallAnalyzer;
+
+    public function __construct(MethodCallAnalyzer $methodCallAnalyzer)
+    {
+        $this->methodCallAnalyzer = $methodCallAnalyzer;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Turns try/catch to expectException() call', [
```
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply packages/PHPUnit/src/Rector/TryCatchToExpectExceptionRector.php]: What does that mean? U prefer explicit over theory

File: phpstan.neon [hunk 3]
```diff
@@ -26,10 +26,14 @@ parameters:
...
         - '#Method Rector\\NodeTypeResolver\\DependencyInjection\\PHPStanServicesFactory::create(.*?)() should return (.*?) but returns object#'
 
+        # false postive - type is set by annotatoin above
+        - '#Access to an undefined property PhpParser\\Node::\$(args|var)#'
+
         # irelevant
         - '#Call to function in_array\(\) with arguments string, array<array<string\|false>> and true will always evaluate to false#'
 
         # known values
+        - '#Parameter \#1 \$methodCallNode of method Rector\\PHPUnit\\Rector\\TryCatchToExpectExceptionRector::renameMethodCallAndKeepFirstArgument\(\) expects PhpParser\\Node\\Expr\\MethodCall, PhpParser\\Node given#'
```
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply phpstan.neon]: PHPStan misses this type.

**Pull request merged**